test/req/steps_test.exs

```diff
@@ -97,36 +111,79 @@ defmodule Req.StepsTest do
       end
     end)
 
-    assert Req.get!(c.url <> "/auth").status == 401
+    directory = System.get_env("NETRC", System.user_home!())
 
-    File.write!("#{c.tmp_dir}/empty_netrc", "")
-    assert Req.get!(c.url <> "/auth", netrc: "#{c.tmp_dir}/empty_netrc").status == 401
+    if File.exists?(Path.join(directory, ".netrc")) do
```

Comment:
this is a bit too clever to have in a test, let's make the test as dumb and deterministic as possible.

Reply:
I agree!

I struggled with this one because without it, the test outcome depends on whether `~/.netrc` is present on the machine running the test. 

Looking at this again I think I can test what I wanted to test with a custom path and that's fine.